main.py

```diff
@@ -356,6 +405,25 @@ def __init__(self, *args):
 		self.actionExit.triggered.connect(self.onExitClicked)
 		self.actionSettings.triggered.connect(self.onSettingsClicked)
 
+		if not "search_for_updates" in self.settings.settings or not "job_id" in self.settings.settings:
+			self.settings.settings["search_for_updates"] = False
+			self.settings.settings["job_id"] = ""
+			self.settings.save()
+
+		if self.settings.settings["search_for_updates"] == True:
+			if os.name == "nt":
+				updater = Updater(self.settings)
+				if not updater.jobs[0]["jobId"] == updater.gsettings.settings["job_id"]:
+					reply = QMessageBox.question(self, 'Update', "A new update for XQEMU is available, would you like to update?\nThe size of the update is: " + str(round(updater.release[0]["size"] / 1e+6, 2)) + " MB", QMessageBox.Yes, QMessageBox.No)
+					if reply == QMessageBox.Yes:
+						dir = os.path.dirname(os.path.abspath(updater.gsettings.settings["xqemu_path"]))
+						if os.path.isdir(dir):
+							updater.update()
+						else:
+							QMessageBox.critical(self, "Update", "The selected XQEMU path doesn't exist.")
+			else:
+				QMessageBox.critical("This is a Windows-only feature.")
```

Comment:
nit: If this block of code was contained in a function, then this can be an early out:

```
if os.name != "nt":
    QMessageBox.critical("This is a Windows-only feature.")
    return

# Code to handle Windows updating
```

I'm not sure if I like this early-out though - just an idea for consideration.